pilot/pkg/bootstrap/server.go

```diff
@@ -398,6 +401,12 @@ func (s *Server) Start(stop <-chan struct{}) error {
 		}()
 	}
 
+	if s.kubeRegistry != nil {
```

Comment:
What is the use case for this? I don't know many apps that report their own startup to events. I am a bit concerned we will just start shoving a bunch of stuff into events for no reason, and it will essentially become logs

Reply:
Agreed, this should not become logs. 

The use case is simple and quite critical - we don't have a good way to know all the Istiods that exist in a multi-cluster system. This will include the address and network, so we can build direct XDS replication - and eventually shard.
The 'connect' events should also include the Istiod instance ID. 

But I completely agree we must have very strict review and limit what is added - my main concern is how well it can 
scale, until we have solid benchmarks we should be super careful. I suspect events are memory-based and too much
spam will affect the health of apiserver - or they may get dropped. 

The intent here is to start gathering data and support small clusters - we still want to integrate with 'real' pubsub.

Reply:
This doesn't solve the use case of finding all Istiods though, as far as I can tell. Events are ephemeral, so you will miss long (where long is a couple minutes) running Istiod instances?

Reply:
Right. It's not ideal - will need to be combined with something else, but for debugging it's quite useful. 
I'm still working on the 'something else'. 

The lack of authentication on event source is a bigger problem - I expect we'll still build XDS connections between Istiod instances, once they discover each other ( even with ephemeral events - i.e. when an Istiod starts up and sends the ephemeral event, other Istiods getting it will connect to it ). The XDS peer-to-peer connections will allow secure exchange of info - including syncing up the connected endpoints or even the store state. Other things like 'nacks' may also help so a nack on a config version could stop other istiods to send. I don't know exactly how far we want to go with this - but a rendezvous is needed, and so far K8S events seem the simplest.